token/js/src/actions/amountToUiAmount.ts

```diff
@@ -28,3 +31,189 @@ export async function amountToUiAmount(
+ * @param lastUpdateTimestamp       Last time the interest rate was updated in seconds
+ * @param initializationTimestamp   Time the interest bearing extension was initialized in seconds
+ * @param preUpdateAverageRate      Interest rate in basis points (hundredths of a percent) before the last update
+ * @param currentRate              Current interest rate in basis points
+ * 
+ * @return Original amount (principle) without interest
+ */
+
+export function uiAmountToAmountWithoutSimulation(
+    uiAmount: string,
+    decimals: number,
+    currentTimestamp: number, // in seconds
+    lastUpdateTimestamp: number,
+    initializationTimestamp: number,
+    preUpdateAverageRate: number,
+    currentRate: number,
+): bigint {
+    Decimal.set({ toExpPos: 24, toExpNeg: -24 })
+    const uiAmountDecimal = new Decimal(uiAmount);
+    const decimalsFactor = new Decimal(10).pow(decimals);
+    const uiAmountScaled = uiAmountDecimal.mul(decimalsFactor);
+   
+    // Calculate pre-update exponent
+    const preUpdateTimespan = new Decimal(lastUpdateTimestamp).minus(initializationTimestamp);
+    const preUpdateNumerator = new Decimal(preUpdateAverageRate).times(preUpdateTimespan);
+    const preUpdateExponent = preUpdateNumerator.div(new Decimal(SECONDS_PER_YEAR).times(ONE_IN_BASIS_POINTS));
+    const preUpdateExp = preUpdateExponent.exp();
+
+    // Calculate post-update exponent
+    const postUpdateTimespan = new Decimal(currentTimestamp).minus(lastUpdateTimestamp);
+    const postUpdateNumerator = new Decimal(currentRate).times(postUpdateTimespan);
+    const postUpdateExponent = postUpdateNumerator.div(new Decimal(SECONDS_PER_YEAR).times(ONE_IN_BASIS_POINTS));
+    const postUpdateExp = postUpdateExponent.exp();
+
+    // Calculate total scale
+    const totalScale = preUpdateExp.times(postUpdateExp);
+
+    // Calculate original principle by dividing the UI amount (principle + interest) by the total scale
+    const originalPrinciple = uiAmountScaled.div(totalScale);
+    return BigInt(originalPrinciple.trunc().toString()); 
+}
+
+/**
+ * Convert a UI amount with interest back to the original UI amount without interest
+ * 
+ * @param connection     Connection to use
+ * @param mint           Mint to get decimals from
+ * @param uiAmount       UI Amount (principle plus continuously compounding interest) to be converted back to original principle
+ * @param programId      SPL Token program account (default: TOKEN_PROGRAM_ID)
+ * 
+ * @return Original UI Amount (principle) without interest
+ */
+export async function uiAmountToAmountForMintWithoutSimulation(
+    connection: Connection,
+    mint: PublicKey,
+    uiAmount: string,
+    programId = TOKEN_PROGRAM_ID,
+): Promise<bigint> {
+    Decimal.set({ toExpPos: 24, toExpNeg: -24 })
+    const mintInfo = await getMint(connection, mint, 'confirmed', programId);
+    const uiAmountScaled = new Decimal(uiAmount).mul(new Decimal(10).pow(mintInfo.decimals));
```

Comment:
nit: same as with the other function, this can get moved under `if (!interestBearingMintConfigState)`